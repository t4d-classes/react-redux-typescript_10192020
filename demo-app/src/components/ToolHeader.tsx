import React from "react";

export type ToolHeaderProps = {
  headerText?: string;
};

export function ToolHeader(props: ToolHeaderProps) {
  return (
    <header>
      <h1>{props.headerText}</h1>
    </header>
  );
}

ToolHeader.defaultProps = {
  headerText: "The Tool",
};
