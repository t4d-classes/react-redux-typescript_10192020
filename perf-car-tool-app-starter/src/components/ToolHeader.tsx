import React from 'react';
import { Typography } from '@material-ui/core';

import { useStyles } from './ToolHeader.styles';

export type ToolHeaderProps = {
  headerText: string;
};

export function ToolHeader({ headerText }: ToolHeaderProps) {
  const classes = useStyles();

  return (
    <header className={classes.header}>
      <Typography variant="h1" className={classes.text}>
        {headerText}
      </Typography>
    </header>
  );
}
