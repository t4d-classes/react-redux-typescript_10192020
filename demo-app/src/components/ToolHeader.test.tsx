import React from 'react';

import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { ToolHeader } from './ToolHeader';

test('snapshot ToolHeader component', () => {
  expect(
    renderer.create(<ToolHeader headerText="The Tool" />).toJSON(),
  ).toMatchSnapshot();
});

describe('ToolHeader component', () => {
  test('renders ToolHeader component', () => {
    const { getByText, getByRole, getAllByRole } = render(
      <ToolHeader headerText="The Tool" />,
    );

    // expect(getByText('The Tool').textContent).toBe('The Tool');
    expect(getByText('The Tool')).toBeInTheDocument();

    expect(getByRole('heading').textContent).toBe('The Tool');
  });
});
