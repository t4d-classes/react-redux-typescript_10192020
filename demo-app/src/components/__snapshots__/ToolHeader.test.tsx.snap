// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot ToolHeader component 1`] = `
<header>
  <h1>
    The Tool
  </h1>
</header>
`;
